### PR TITLE
Rework NavMenu.tsx and Layout.tsx components

### LINKED ISSUE #17351: [new UD] Re-vamp NavMenu.tsx and Layout.tsx components
For a better project structure and code organization:

- Move `NavMenu.tsx` and `Layout.tsx` into `/src/components/layout` directory
- From `NavMenu.tsx`, extract code related to the page header as a separate component - `/src/components/layout/Header.tsx`
- Rename `NavMenu.tsx` to `Sidebar.tsx`
- Re-work `Layout.tsx` to create the page layout from sidebar, header and main page content

### PR DESCRIPTION
**Pull request by akurinnoy**
resolves https://github.com/eclipse/che/issues/17351

The umbrella CQ for new dev dependencies: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22418

### CODE REVIEW

File: src/components/Layout/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    const userEmail = this.props.user?.email || '';
+    const userName = this.props.user?.name || '';
+
+    const className = this.state.isVisible ? styles.headerShow : styles.headerHide;
+
+    return (
+      <PageHeader
+        className={className}
+        logo={logo}
+        logoProps={{ href: 'https://www.eclipse.org/che/', target: '_blank' }}
```
Comment (olexii4) [inline src/components/Layout/Header.tsx]: It should be the [helpPath](https://github.com/che-incubator/che-dashboard-next/blob/master/assets/branding/product.json#L9) variable from the product.json

File: src/components/Layout/HeaderTools.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    const toggleButton = this.buildToggleButton();
+    const dropdownItems = this.buildDropdownItems();
+
+    return (
+      <PageHeaderTools>
+        <PageHeaderToolsGroup>
+          <PageHeaderToolsItem>
+            <Dropdown
+              isPlain
+              position="right"
```
Comment (olexii4) [inline src/components/Layout/HeaderTools.tsx:121]: ```"``` -> ```'```

File: src/components/Layout/Navigation/RecentList.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+import NavigationRecentItem from './RecentItem';
+import { NavigationItemObject, NavigationRecentItemObject } from '.';
+import NavigationMainItem from './MainItem';
+
+import styles from './index.module.css';
+
+function buildCreateWorkspaceItem(): React.ReactElement {
+  const item: NavigationItemObject = {
+    to: '/get-started#custom-workspace',
```
Comment (olexii4) [inline src/components/Layout/Navigation/RecentList.tsx]: It should be getting from ```enum ROUTE```
https://github.com/che-incubator/che-dashboard-next/pull/48/files#diff-510ca6cdc30d8b32c0054feb11b8e8f2R15

Comment (olexii4) [reply src/components/Layout/Navigation/RecentList.tsx]: WDYT?

File: .deps/dev.md [hunk 4]
```diff
@@ -616,10 +672,14 @@
 | [`node-modules-regexp@1.0.0`](https://github.com/jamestalmage/node-modules-regexp.git) | MIT | clearlydefined |
 | [`node-notifier@7.0.1`](git+ssh://git@github.com/mikaelbr/node-notifier.git) | MIT |  |
 | [`node-object-hash@1.4.2`](git@github.com:SkeLLLa/node-object-hash.git) | ISC | clearlydefined |
+| [`node-releases@1.1.59`](git+https://github.com/chicoxyzzy/node-releases.git) | MIT | clearlydefined |
 | [`normalize-package-data@2.5.0`](git://github.com/npm/normalize-package-data.git) | BSD-2-Clause |  |
 | [`normalize-path@3.0.0`](https://github.com/jonschlinkert/normalize-path.git) | MIT | [CQ17757](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=17757) |
+| [`normalize-range@0.1.2`](https://github.com/jamestalmage/normalize-range.git) | MIT | [CQ18717](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18717) |
+| [`normalize-selector@0.2.0`](git://github.com/getify/normalize-selector.git) | MIT |  |
```
Comment (olexii4) [inline .deps/dev.md:680]: I found 6 new dependencies. Could you provide links for all needed CQs?

File: .deps/dev.md [hunk 5]
```diff
@@ -680,13 +741,25 @@
...
 | [`postcss-modules-scope@2.2.0`](https://github.com/css-modules/postcss-modules-scope.git) | ISC | clearlydefined |
 | [`postcss-modules-values@3.0.0`](git+https://github.com/css-modules/postcss-modules-values.git) | ISC | clearlydefined |
+| [`postcss-reporter@6.0.1`](https://github.com/postcss/postcss-reporter.git) | MIT | clearlydefined |
+| [`postcss-resolve-nested-selector@0.1.1`](Unknown) | MIT | clearlydefined |
+| [`postcss-safe-parser@4.0.2`](https://github.com/postcss/postcss-safe-parser.git) | MIT | clearlydefined |
+| [`postcss-sass@0.4.4`](https://github.com/AleshaOleg/postcss-sass.git) | MIT | clearlydefined |
+| [`postcss-scss@2.1.1`](https://github.com/postcss/postcss-scss.git) | MIT | clearlydefined |
 | [`postcss-selector-parser@6.0.2`](https://github.com/postcss/postcss-selector-parser.git) | MIT | clearlydefined |
+| [`postcss-sorting@5.0.1`](https://github.com/hudochenkov/postcss-sorting.git) | MIT | clearlydefined |
+| [`postcss-syntax@0.36.2`](git+https://github.com/gucong3000/postcss-syntax.git) | MIT |  |
```
Comment (olexii4) [inline .deps/dev.md:760]: Could you provide a link for the needed CQ?

File: .deps/dev.md [hunk 6]
```diff
@@ -719,10 +793,14 @@
 | [`regex-not@1.0.2`](https://github.com/jonschlinkert/regex-not.git) | MIT | [CQ18717](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18717) |
 | [`regexpp@3.1.0`](git+https://github.com/mysticatea/regexpp.git) | MIT | clearlydefined |
 | [`relateurl@0.2.7`](git://github.com/stevenvachon/relateurl.git) | MIT | clearlydefined |
+| [`remark-parse@8.0.2`](https://github.com/remarkjs/remark/tree/master/packages/remark-parse) | MIT |  |
```
Comment (olexii4) [inline .deps/dev.md:796]: Could you provide a link for the needed CQ?

File: .deps/dev.md [hunk 7]
```diff
@@ -719,10 +793,14 @@
...
+| [`remark-stringify@8.1.0`](https://github.com/remarkjs/remark/tree/master/packages/remark-stringify) | MIT |  |
```
Comment (olexii4) [inline .deps/dev.md:797]: Could you provide a link for the needed CQ?

File: .deps/dev.md [hunk 8]
```diff
@@ -839,6 +930,10 @@
 | [`toidentifier@1.0.0`](https://github.com/component/toidentifier.git) | MIT | clearlydefined |
 | [`tough-cookie@2.5.0`](git://github.com/salesforce/tough-cookie.git) | BSD-3-Clause | clearlydefined |
 | [`tr46@2.0.2`](https://github.com/jsdom/tr46) | MIT | clearlydefined |
+| [`trim-newlines@3.0.0`](https://github.com/sindresorhus/trim-newlines.git) | MIT | clearlydefined |
+| [`trim-trailing-lines@1.1.3`](https://github.com/wooorm/trim-trailing-lines.git) | MIT | clearlydefined |
+| [`trim@0.0.1`](Unknown) | MIT |  |
```
Comment (olexii4) [inline .deps/dev.md:935]: Could you provide a link for the needed CQ?

File: src/components/Layout/HeaderTools.tsx (hunk 2 above)
Comment (akurinnoy) [reply src/components/Layout/HeaderTools.tsx:121]: As far as it's html let's leave double quotes there.

File: .deps/dev.md [hunk 9]
```diff
@@ -900,16 +1006,17 @@
 | [`worker-farm@1.7.0`](https://github.com/rvagg/node-worker-farm.git) | MIT | clearlydefined |
 | [`wrap-ansi@6.2.0`](https://github.com/chalk/wrap-ansi.git) | MIT | clearlydefined |
 | [`wrappy@1.0.2`](https://github.com/npm/wrappy) | ISC | [CQ16925](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=16925) |
-| [`write-file-atomic@2.4.3`](git@github.com:iarna/write-file-atomic.git) | ISC | clearlydefined |
+| [`write-file-atomic@3.0.3`](git://github.com/npm/write-file-atomic.git) | ISC | clearlydefined |
 | [`write-json-file@2.3.0`](https://github.com/sindresorhus/write-json-file.git) | MIT | [CQ18717](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18717) |
 | [`write@1.0.3`](https://github.com/jonschlinkert/write.git) | MIT | clearlydefined |
 | [`ws@6.2.1`](https://github.com/websockets/ws.git) | MIT | clearlydefined |
+| [`x-is-string@0.1.0`](git://github.com/Matt-Esch/x-is-string.git) | MIT |  |
```
Comment (olexii4) [inline .deps/dev.md:1013]: Could you provide a link for the needed CQ?

File: src/components/Layout/Navigation/RecentList.tsx (hunk 3 above)
Comment (akurinnoy) [reply src/components/Layout/Navigation/RecentList.tsx]: good point, thanks

File: .deps/dev.md (hunk 4 above)
Comment (akurinnoy) [reply .deps/dev.md:680]: no, as it's a generated file.

File: .deps/dev.md (hunk 5 above)
Comment (akurinnoy) [reply .deps/dev.md:760]: it's a generated file. All changes added manually will be lost after running license tools.

File: .deps/dev.md (hunk 6 above)
Comment (akurinnoy) [reply .deps/dev.md:796]: no

File: .deps/dev.md (hunk 7 above)
Comment (akurinnoy) [reply .deps/dev.md:797]: no

File: .deps/dev.md (hunk 4 above)
Comment (akurinnoy) [reply .deps/dev.md:680]: It seems I didn't get your question. 
Did you mean to add the CQ link to the `dev.md` or just asking if I created CQs for a new dependency?

Comment (olexii4) [reply .deps/dev.md:680]: I asking to create CQs for all new dependencies and add links as a comment near each new line without approval for manual checking(it is not necessary, just a variant for solving the problem). In this case, it should be one umbrella issue.

Comment (akurinnoy) [reply .deps/dev.md:680]: OK, thanks, now I got it.
I'll check and create CQs if necessary and update the PR description.

**Comment by olexii4** [comment]
One more problem: I saw a regression. We need to create a sub-issue or fix it in this PR.
Screenshot from 2020-07-23 12-01-06

url '.../#/get-started##get-started'

It reproduces in the case when clicking on 'Workspaces', then 'Get Started Page'.

**Comment by olexii4** [comment]
In general, it is a good job. Thanks.

File: src/components/Layout/HeaderTools.tsx (hunk 2 above)
Comment (olexii4) [reply src/components/Layout/HeaderTools.tsx:121]: ok

**Review by olexii4 — APPROVED** [review]

**Pull request merged**